# Capstone Project - The Battle of Neighborhoods (Week 1 & 2)

This capstone project will be graded by your peers. This capstone project is worth 70% of your total grade. The project will be completed over the course of 2 weeks. Week 1 submissions will be worth 30% whereas week 2 submissions will be worth 40% of your total grade.

For this week, you will required to submit the following:

A description of the problem and a discussion of the background. (15 marks)
A description of the data and how it will be used to solve the problem. (15 marks)
For the second week, the final deliverables of the project will be:

A link to your Notebook on your Github repository, showing your code. (15 marks)
A full report consisting of all of the following components (15 marks):
Introduction where you discuss the business problem and who would be interested in this project.
Data where you describe the data that will be used to solve the problem and the source of the data.
Methodology section which represents the main component of the report where you discuss and describe any exploratory data analysis that you did, any inferential statistical testing that you performed, if any, and what machine learnings were used and why.
Results section where you discuss the results.
Discussion section where you discuss any observations you noted and any recommendations you can make based on the results.
Conclusion section where you conclude the report.
3. Your choice of a presentation or blogpost. (10 marks)

<B> NOTE: The following code will have both the report and the code compiled to minimize the effort to create an additonal documentation. The objective of this notebook is to Document simulatneously with code.

## Coursera Capstone - Report

<b> CONTENT

Introduction Section :

1.1 Discussion of the "backgroung situation" leading to the problem at hand:

1.2 Problem to be resolved

1.3 Audience for this project.

Data Section:

2.1 Data of Current Situation (current residence place)

2.2 Data required to resolve the problem

2.3 Data sources and data manipulation

Methodology section :

3.1 Process steps and strategy to resolve the problem

3.2 Data Science Methods, machine learing, mapping tools and exploratory data analysis.

Results section

Discussion of the results and how they help to take a decision.

Discussion section

Elaboration and discussion on any observations and/or recommendations for improvement.

Conclusion section

Desicison taken and Report Conclusion.

## INTRODUCTION

### 1.1 BACKGROUND

I am a Jr.data scientist currently interning in Dallas, Tx. I currently live in Addison, in texas people drive cars and there isnt a feasible public transportation as the car fare is relatively cheaper. Likewise, I live close to belt line road that has many ammenities such as international cousine restaurants, cafes, food shops and entertainment. I have been offered a great opportunity to work in Manhattan, NY. Although, I am very excited about it, I am a bit stress toward the process to secure a comparable place to live in Manhattan. Therefore, I decided to apply the learned skills during the Coursera course to explore ways to make sure my decision is factual and rewarding. Of course, there are alternatives to achieve the answer using available Google and Social media tools, but it rewarding doing it myself with learned tools.

### 1.2 PROBLEM TO BE SOLVED

The challenge to resolve is being able to find a rental apartment unit in Manhattan NY that offers similar characteristics and benefits to my current situation. Therefore, in order to set a basis for comparison, I want to find a renta unit subject to the following conditions:

1. Apartment with min 2 bedrooms with monthly rent not to exceed US$7000/month
2. Unit located within walking distance (<= 1.0 mile) from a subway metro station in Manhattan
3. Area with ammenities and venues similar to the ones described for current location ( See item 2.1)

### 1.3 INTERESTED AUDIENCE

I believe this is a relevant project for a person considering moving to a major city in the US, since the approach and methodologies used here are applicable in all cases. The use of FourSquare data and mapping techniques combined with data analysis will help resolve the key questions arisen. Lastly, this project is a good practical case toward the development of Data Science skills.

## DATA SECTION

### 2.1 Data of Current Location

I Currently reside in the Addison, Tx that camoes under Dallas. I use Foursquare to identify the venues around the area of residence which are then shown in the Dallas map shown in methodology and execution in section 3.0 . It serves as a reference for comparison with the desired future location in Manhattan,NY

### 2.2 Data Required to resolve the problem.

In order to make a good choice of a similar apartment in Manhattan NY, the following data is required: List/Information on neighborhoods form Manhattan with their Geodata ( latitud and longitud. List/Information about the subway metro stations in Manhattan with geodata. Listed apartments for rent in Manhattan area with descriptions ( how many beds, price, location, address) Venues and ammenities in the Manhattan neighborhoods (e.g. top 10) 2.3 sources and manipulation The list of Manhattan neighborhoods is worked out during LAb exercise during the course. A csv file was created which will be read in order to create a dataframe and its mapping. The csv file 'mh_neigh_data.csv' has the following below data structure. The file will be directly read to the Jupiter Notebook for convenience and space savings. The clustering of neighborhoods and mapping will be shown however. An algorythm was used to determine the geodata from Nominatim . The actual algorythm coding may be shown in 'markdown' mode becasues it takes time to run.

sample of mh_neigh_data:

Borough, Neighborhood, Latitude, Longitude

1. Manhattan, Turtle Bay, 40.752042, -73.967708
2. Manhattan, Tudor City, 40.746917, -73.971219
      

A list of Manhattan subway metro stops was complied in Numbers (Apple excel) and it was complemeted with wikipedia data (link-1) and information from NY Transit authority and Google maps (Link-2) for a final consolidated list of subway stops names and their address. The geolocation was obtained via an algorythm using Nominatim. Details will be shown in the execution of methodolody in section 3.0. The subway csv file is ''MH_subway.csv'" and the data structure is: mhsub.tail(): sub_station sub_address lat long

1. 190 Street Subway Station, Bennett Ave, New York, NY 10040, USA, 40.858113, -73.932983
2. 59 St-Lexington Av Station, E 60th St, New York, NY 10065, USA, 40.762259, -73.966271

A list of places for rent was collected by web-browsing real estate companies in Manhattan : (Link-3) A csv file was compiled with the rental place that indicated: areas of Manhattan, address, number of beds, area and monthly rental price. The csv file "nnnn.csv" had the following below structure. An algorythm was used to create all the geodata using Nominatim, as shown in section 3.0. The actual algorythm coding may be shown in 'markdown' mode becasues it takes time to run. With the use of geolocator = Nominatim() , it was possible to determine the latitude and longiude for the subway metro locations as well as for the geodata for each rental place listed. The loop algorythms used are shown in the execution of data in section 3.0 "Great_circle" function from geolocator was used to calculate distances between two points , as in the case to calculate average rent price for units around each subway station and at 1.6 km radius. Foursquare is used to find the avenues at Manhattan neighborhoods in general and a cluster is created to later be able to search for the venues depending of the location shown.

Link-1 : https://en.wikipedia.org/wiki/List_of_New_York_City_Subway_stations_in_Manhattan.

Link-2 : https://www.google.com/maps/search/manhattan+subway+metro+stations/@40.7837297,-74.1033043,11z/data=!3m1!4b1.

Link-3 : http://www.rentmanhattan.com/index.cfm?page=search&state=results https://www.nestpick.com/search?city=new-york&page=1&order=relevance&district=manhattan&gclid=CjwKCAiAjNjgBRAgEiwAGLlf2hkP3A-cPxjZYkURqQEswQK2jKQEpv_MvKcrIhRWRzNkc_r-fGi0lxoCA7cQAvD_BwE&type=apartment&display=list https://www.realtor.com/apartments/Manhattan_NY

### 2.4 How the data will be used to solve the problem.

The data will be used as follows: Use Foursquare and geopy data to map top 10 venues for all Manhattan neighborhoods and clustered in groups ( as per Course LAB) Use foursquare and geopy data to map the location of subway metro stations , separately and on top of the above clustered map in order to be able to identify the venues and ammenities near each metro station, or explore each subway location separately Use Foursquare and geopy data to map the location of rental places, in some form, linked to the subway locations. create a map that depicts, for instance, the average rental price per square ft, around a radious of 1.0 mile around each subway station - or a similar metrics. I will be able to quickly point to the popups to know the relative price per subway area. Addresses from rental locations will be converted to geodata( lat, long) using Geopy-distance and Nominatim. Data will be searched in open data sources if available, from real estate sites if open to reading, libraries or other government agencies such as Metro New York MTA, etc.

### 2.5 Data Mapping

The following maps were created to facilitate the analysis and the choice of the palace to live. Manhattan map of Neighborhoods manhattan subway metro locations Manhattan map of places for rent Manhattan map of clustered venues and neighborhoods Combined maps of Manhattan rent places with subway locations Combined maps of Manhattan rent places with subway locations and venues clusters

## 3. METHODOLOGY SECTION

This section represents the main component of the report where the data is gathered, prepared for analysis. The tools described are used here and the Notebook cells indicates the execution of steps.

### STRATEGY FOR THE ANALYSIS

The strategy is based on mapping the above described data in section 2.0, in order to facilitate the choice of at least two candidate places for rent. The choice is made based on the demands imposed : location near a subway, rental price and similar venues to Singapore. This visual approach and maps with popups labels allow quick identification of location, price and feature, thus making the selection very easy.

The procesing of these DATA and its mapping will allow to answer the key questions to make a decision:

what is the cost of available rental places that meet the demands?
what is the cost of rent around a mile radius from each subway metro station?
what is the area of Manhattan with best rental pricing that meets criteria established?
What is the distance from work place ( Park Ave and 53 rd St) and the tentative future rental home?
What are the venues of the two best places to live? How the prices compare?
How venues distribute among Manhattan neighborhoods and around metro stations?
Are there tradeoffs between size and price and location?
Any other interesting statistical data findings of the real estate and overall data.

## METHODOLOGY EXECUTION 

<b> Data Mapping.


Dallas Map for the Current residence and venues near the neighborhood for comparison to future Manhattan renting place

## IMPORTING PACKAGES

In [ ]:
import requests # library to handle requests
import pandas as pd # library for data analsysis
import numpy as np # library to handle data in a vectorized manner
import random # library for random number generation
from geopy.geocoders import Nominatim # module to convert an address into latitude and longitude values

# libraries for displaying images
from IPython.display import Image 
from IPython.core.display import HTML 
    
# tranforming json file into a pandas dataframe library
from pandas.io.json import json_normalize

# plotting library
import folium
import plotly.express as px
from folium import plugins
import matplotlib.cm as cm
import matplotlib.colors as colors
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

#Scikit-Packages
from sklearn.cluster import KMeans

print('Folium installed')
print('Libraries imported.')

In [2]:
# Addison, TX
address = '14651 Dallas Pkwy, Dallas, TX'
geolocator = Nominatim()
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
neighborhood_latitude = latitude
neighborhood_longitude = longitude
print('The geograpical coordinate of Dallas home are {}, {}.'.format(latitude, longitude))

C:\Users\gteja\AppData\Local\Continuum\anaconda3\lib\site-packages\ipykernel_launcher.py:3: DeprecationWarning: Using Nominatim with the default "geopy/1.20.0" `user_agent` is strongly discouraged, as it violates Nominatim's ToS https://operations.osmfoundation.org/policies/nominatim/ and may possibly cause 403 and 429 HTTP errors. Please specify a custom `user_agent` with `Nominatim(user_agent="my-application")` or by overriding the default `user_agent`: `geopy.geocoders.options.default_user_agent = "my-application"`. In geopy 2.0 this will become an exception.
  This is separate from the ipykernel package so we can avoid doing imports until


The geograpical coordinate of Dallas home are 32.9471863265306, -96.8229930408163.


<b>Dial FourSquare to find venues around current residence in Dallas

In [3]:
CLIENT_ID = 'NY2BASND4KRZPNJLLX1OFJUYRVDKBO4VMWA0BD30YOU53KBT' # your Foursquare ID
CLIENT_SECRET = 'FSNLUGQ0BNKF20TFAVE4LYGMXW4VHMXA1QJCYVVHPO0NCNLL' # your Foursquare Secret
VERSION = '20180604'
LIMIT = 100
radius = 500
#print('Your credentails:')
#print('CLIENT_ID: ' + CLIENT_ID)
#print('CLIENT_SECRET:' + CLIENT_SECRET)

In [4]:
# create URL
url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
    CLIENT_ID, 
    CLIENT_SECRET, 
    VERSION, 
    neighborhood_latitude, 
    neighborhood_longitude, 
    radius, 
    LIMIT)
url # display URL

'https://api.foursquare.com/v2/venues/explore?&client_id=NY2BASND4KRZPNJLLX1OFJUYRVDKBO4VMWA0BD30YOU53KBT&client_secret=FSNLUGQ0BNKF20TFAVE4LYGMXW4VHMXA1QJCYVVHPO0NCNLL&v=20180604&ll=32.9471863265306,-96.8229930408163&radius=500&limit=100'

In [5]:
# results display is hidden for report simplification 
results = requests.get(url).json()
#results

<b>Custom function that extracts the category of the venue - borrow from the Foursquare lab.

In [6]:
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

In [7]:
venues = results['response']['groups'][0]['items']
dlnearby_venues = json_normalize(venues) # flatten JSON
# filter columns
filtered_columns = ['venue.name', 'venue.categories', 'venue.location.lat', 'venue.location.lng']
dlnearby_venues = dlnearby_venues.loc[:, filtered_columns]
# filter the category for each row
dlnearby_venues['venue.categories'] = dlnearby_venues.apply(get_category_type, axis=1)
# clean columns
dlnearby_venues.columns = [col.split(".")[-1] for col in dlnearby_venues.columns]

dlnearby_venues.shape

(17, 4)

In [8]:
# Venues near current Dallas residence place
dlnearby_venues.head(10)

name               categories  \
0                           Lawry's The Prime Rib               Steakhouse   
1                    AMC Village on the Parkway 9            Movie Theater   
2                             Hopdoddy Burger Bar             Burger Joint   
3                                      Yard House      American Restaurant   
4                      A Feel Good Experience Spa                      Spa   
5                       Lazy Dog Restaurant & Bar  New American Restaurant   
6  Dallas/Addison Marriott Quorum by the Galleria                    Hotel   
7                                       Grand Spa                      Spa   
8                               Pluckers Wing Bar              Wings Joint   
9                         RA Sushi Bar Restaurant         Asian Restaurant   

         lat        lng  
0  32.945278 -96.823866  
1  32.950495 -96.820608  
2  32.950926 -96.821021  
3  32.950553 -96.819781  
4  32.950441 -96.822471  
5  32.951308 -96.820927  
6  32.951297 -96.824301  
7  32.951507 -96.821914  
8  32.950870 -96.820381  
9  32.950733 -96.821090

<b> Map of residence place with Neighborhood venues

In [15]:
# create map of Dallas place  using latitude and longitude values
fig = px.scatter_mapbox(dlnearby_venues, lat= 'lat', lon = 'lng', 
                       hover_data=['name', 'categories', 'lat', 'lng'],
                        color_discrete_sequence=["fuchsia"], zoom=15, height=300)
fig.update_layout(mapbox_style="open-street-map")
fig.update_layout(margin={"r":0,"t":0,"l":0,"b":0})
fig.show()

In [12]:
# Reading the csv file with clustered neighborhoods of Manhattan with geodata
mh_data  = pd.read_csv('datasets/mh_neigh_data.csv') 
#mh_data.head()

In [ ]:
#mh_data.info()

<b>Manhattan Borough neighborhoods of top 10 clustered venues

In [13]:
mh_merged = pd.read_csv('datasets/manhattan_merged.csv')
mh_merged.head()

Borough        Neighborhood   Latitude  Longitude  Cluster Labels  \
0  Manhattan         Marble Hill  40.876551 -73.910660               2   
1  Manhattan           Chinatown  40.715618 -73.994279               2   
2  Manhattan  Washington Heights  40.851903 -73.936900               4   
3  Manhattan              Inwood  40.867684 -73.921210               3   
4  Manhattan    Hamilton Heights  40.823604 -73.949688               0   

  1st Most Common Venue 2nd Most Common Venue 3rd Most Common Venue  \
0           Coffee Shop        Discount Store           Yoga Studio   
1    Chinese Restaurant          Cocktail Bar    Dim Sum Restaurant   
2                  Café                Bakery     Mobile Phone Shop   
3    Mexican Restaurant                Lounge           Pizza Place   
4    Mexican Restaurant           Coffee Shop                  Café   

  4th Most Common Venue  5th Most Common Venue 6th Most Common Venue  \
0            Steakhouse        Supplement Shop        Tennis Stadium   
1   American Restaurant  Vietnamese Restaurant    Salon / Barbershop   
2           Pizza Place         Sandwich Place                  Park   
3                  Café               Wine Bar                Bakery   
4         Deli / Bodega            Pizza Place          Liquor Store   

  7th Most Common Venue      8th Most Common Venue 9th Most Common Venue  \
0            Shoe Store                        Gym                  Bank   
1          Noodle House                     Bakery       Bubble Tea Shop   
2                   Gym  Latin American Restaurant      Tapas Restaurant   
3   American Restaurant                       Park    Frozen Yogurt Shop   
4     Indian Restaurant           Sushi Restaurant        Sandwich Place   

  10th Most Common Venue  
0     Seafood Restaurant  
1         Ice Cream Shop  
2     Mexican Restaurant  
3     Spanish Restaurant  
4            Yoga Studio

<b> Lets Map the top 10 clustered venues Manhattan  

In [39]:
# create map of Manhattan using latitude and longitude values from Nominatim
latitude= 40.7308619
longitude= -73.9871558 

# add markers to the map
fig = px.scatter_mapbox(mh_merged, lat= 'Latitude', lon = 'Longitude', color = 'Cluster Labels', labels = 'Neighborhood',
                        #hover_data=['Neighborhood', 'Cluster Labels'],
                        color_discrete_sequence= px.colors.sequential.Viridis, 
                        zoom=7, height=300)
fig.update_layout(mapbox_style="open-street-map")
fig.update_layout(margin={"r":0,"t":0,"l":0,"b":0})
fig.show()

# add markers for rental places to map
fig = px.scatter_mapbox(mh_data, lat= 'Latitude', lon = 'Longitude', labels = 'Neighborhood',color = 'Neighborhood',
                        color_discrete_sequence=px.colors.sequential.Viridis, zoom=7, height=300)
fig.update_layout(mapbox_style="open-street-map")
fig.update_layout(margin={"r":0,"t":0,"l":0,"b":0})
fig.show()

<b> Lets examine particular clusters. 
    Further, after carefull examination of the data. The ideal plcae would be something Cluster#3 that is similar to place in       addison

In [89]:
## k_num is the cluster number to explore
k_num = 3
mh_merged.loc[mh_merged['Cluster Labels'] == k_num, 
                     mh_merged.columns[[1] + list(range(5, mh_merged.shape[1]))]]

Neighborhood 1st Most Common Venue 2nd Most Common Venue  \
3               Inwood    Mexican Restaurant                Lounge   
5       Manhattanville         Deli / Bodega    Italian Restaurant   
10          Lenox Hill      Sushi Restaurant    Italian Restaurant   
12     Upper West Side    Italian Restaurant                   Bar   
16         Murray Hill        Sandwich Place                 Hotel   
17             Chelsea           Coffee Shop    Italian Restaurant   
18   Greenwich Village    Italian Restaurant      Sushi Restaurant   
27            Gramercy    Italian Restaurant            Restaurant   
29  Financial District           Coffee Shop                 Hotel   
31                Noho    Italian Restaurant     French Restaurant   
32        Civic Center  Gym / Fitness Center                Bakery   
35          Turtle Bay    Italian Restaurant           Coffee Shop   
36          Tudor City                  Café                  Park   
38            Flatiron    Italian Restaurant   American Restaurant   

     3rd Most Common Venue          4th Most Common Venue  \
3              Pizza Place                           Café   
5       Seafood Restaurant             Mexican Restaurant   
10             Coffee Shop           Gym / Fitness Center   
12                  Bakery  Vegetarian / Vegan Restaurant   
16     Japanese Restaurant           Gym / Fitness Center   
17          Ice Cream Shop                         Bakery   
18       French Restaurant                 Clothing Store   
27  Thrift / Vintage Store                   Cocktail Bar   
29                     Gym                      Wine Shop   
31            Cocktail Bar                      Gift Shop   
32      Italian Restaurant                   Cocktail Bar   
35              Steakhouse                       Wine Bar   
36             Pizza Place             Mexican Restaurant   
38                     Gym           Gym / Fitness Center   

   5th Most Common Venue          6th Most Common Venue 7th Most Common Venue  \
3               Wine Bar                         Bakery   American Restaurant   
5       Sushi Restaurant                    Beer Garden           Coffee Shop   
10           Pizza Place                   Burger Joint         Deli / Bodega   
12     Indian Restaurant                    Coffee Shop        Cosmetics Shop   
16           Coffee Shop             Salon / Barbershop          Burger Joint   
17             Nightclub                        Theater           Art Gallery   
18    Chinese Restaurant                           Café     Indian Restaurant   
27            Bagel Shop                    Coffee Shop           Pizza Place   
29            Steakhouse                            Bar    Italian Restaurant   
31             Bookstore                  Grocery Store    Mexican Restaurant   
32     French Restaurant                 Sandwich Place           Coffee Shop   
35      Sushi Restaurant                          Hotel          Noodle House   
36      Greek Restaurant               Sushi Restaurant                 Hotel   
38           Yoga Studio  Vegetarian / Vegan Restaurant                Bakery   

   8th Most Common Venue 9th Most Common Venue 10th Most Common Venue  
3                   Park    Frozen Yogurt Shop     Spanish Restaurant  
5     Falafel Restaurant            Bike Trail        Other Nightlife  
10                   Gym   Sporting Goods Shop        Thai Restaurant  
12              Wine Bar    Mexican Restaurant       Sushi Restaurant  
16     French Restaurant                   Bar     Italian Restaurant  
17    Seafood Restaurant   American Restaurant                  Hotel  
18                Bakery    Seafood Restaurant      Electronics Store  
27    Mexican Restaurant         Grocery Store              Wine Shop  
29           Pizza Place                  Park   Gym / Fitness Center  
31                 Hotel      Sushi Restaurant            Coffee Shop  
32                   Gym           

<b>Several Manhattan real estate webs were webscrapped to collect rental data. The resut was summarized in a csv file for direct reading, in order to consolidate the proces. The initial data for 144 apartment did not have the latitude and longitude data (NaN) but the information was established in the following cell using an algorythm and Nominatim.

In [170]:
mh_rent=pd.read_csv('datasets/MH_flats_price.csv')
mh_rent.head(10)

Address             Area  Price_per_ft2  Rooms  Area-ft2  \
0      West 105th Street  Upper West Side           2.94    5.0      3400   
1       East 97th Street  Upper East Side           3.57    3.0      2100   
2      West 105th Street  Upper West Side           1.89    4.0      2800   
3            CARMINE ST.     West Village           3.03    2.0      1650   
4        171 W 23RD ST.           Chelsea           3.45    2.0      1450   
5        76 CARMINE ST.      West Village           3.56    2.0      1350   
6                 E 91ST  Upper East Side           3.21    2.0      1450   
7  333 east 81st street   Upper East Side           1.85    3.0      2430   
8       EAST 88TH STREET  Upper East Side           3.21    2.0      1400   
9       East 77th Street  Upper East Side           5.00    1.0       900   

   Rent_Price  Lat  Long  
0       10000  NaN   NaN  
1        7500  NaN   NaN  
2        5300  NaN   NaN  
3        5000  NaN   NaN  
4        5000  NaN   NaN  
5        4800  NaN   NaN  
6        4650  NaN   NaN  
7        4500  NaN   NaN  
8        4500  NaN   NaN  
9        4500  NaN   NaN

<b>Obtain geodata ( lat,long) for each rental place in Manhattan with Nominatim¶

Data was stored in a csv file for simplifaction report purposes and saving code processing time in future.
This coding section was 'markedown' for the report because its execution takes few minutes . Terefore, the csv saved will be be just read directly in the following cell.
for n in range(len(mh_rent)):
address= mh_rent['Address'][n] address=(mh_rent['Address'][n]+ ' , '+' Manhattan NY ') geolocator = Nominatim() location = geolocator.geocode(address) latitude = location.latitude longitude = location.longitude mh_rent['Lat'][n]=latitude mh_rent['Long'][n]=longitude

#print(n,latitude,longitude)
time.sleep(2)

print('Geodata completed')

save dataframe to csv file

In [171]:
mh_rent=pd.read_csv('datasets/MH_rent_latlong.csv')
mh_rent.head(10)

Address             Area  Price_per_ft2  Rooms  Area-ft2  \
0      West 105th Street  Upper West Side           2.94    5.0      3400   
1       East 97th Street  Upper East Side           3.57    3.0      2100   
2      West 105th Street  Upper West Side           1.89    4.0      2800   
3            CARMINE ST.     West Village           3.03    2.0      1650   
4        171 W 23RD ST.           Chelsea           3.45    2.0      1450   
5        76 CARMINE ST.      West Village           3.56    2.0      1350   
6                 E 91ST  Upper East Side           3.21    2.0      1450   
7  333 east 81st street   Upper East Side           1.85    3.0      2430   
8       EAST 88TH STREET  Upper East Side           3.21    2.0      1400   
9       East 77th Street  Upper East Side           5.00    1.0       900   

   Rent_Price        Lat       Long  
0       10000  40.799771 -73.966213  
1        7500  40.788585 -73.955277  
2        5300  40.799771 -73.966213  
3        5000  40.730523 -74.001873  
4        5000  40.744118 -73.995299  
5        4800  40.729631 -74.004440  
6        4650  40.781059 -73.949421  
7        4500  40.774324 -73.952836  
8        4500  40.779176 -73.950794  
9        4500  40.772969 -73.958253

## INFERENTIAL STATISTICS

<b> Data Visulaization of Manhattan Apartment Rent Price Statistics

In [57]:
fig = px.histogram(mh_rent, x="Rent_Price", marginal="box")
fig.show()

It can be observed from the plot that the median is close to $8000 which is a lot higher than in dallas. 

In [58]:
fig = px.histogram(mh_rent, x="Price_per_ft2", marginal="box")
fig.show()

In [63]:
fig = px.box(mh_rent, x="Rooms", y = "Rent_Price",points="all")
fig.show()

<b>Funnel plot for Area and Rooms

In [70]:
from plotly import graph_objects as go

fig = go.Figure(go.Funnel(
    y = mh_rent['Area'],
    x = mh_rent['Rooms'], orientation = 'v'))

fig.show()

<b>Lets map the places that can be rented in Manhattan.The plot will indicate the address and the monthly price for rent thus making it convenient to select the target appartment with the price condition estimated. 

In [90]:
fig = px.scatter_mapbox(mh_rent, lat= 'Lat', lon = 'Long', labels = 'Rent_Price', color = 'Address',
                        hover_data=['Address'],
                        color_discrete_sequence= px.colors.cyclical.IceFire, 
                        zoom=10, height=500)
fig.update_layout(mapbox_style="open-street-map")
fig.update_layout(margin={"r":0,"t":0,"l":0,"b":0})
fig.show()

<b>lets map the places for rent and the cluster of venues of Manhattan.
Now, one can point to a rental place for price and address location information while knowing the cluster venues around it.
This is an insightful way to explore rental possibilites under $7000.

In [113]:
mh_map_merged = mh_merged.merge(mh_rent, left_on= 'Neighborhood', right_on = 'Area', how = 'outer')
mh_map_merged = mh_map_merged[mh_map_merged['Rent_Price'] <=7000]

In [114]:
fig = px.scatter_mapbox(mh_map_merged, lat= 'Lat', lon = 'Long', color = 'Rent_Price', labels = 'Cluster Labels',
                        hover_data=['Neighborhood', 'Cluster Labels','Rent_Price'],
                        color_discrete_sequence= px.colors.sequential.Viridis, 
                        zoom=10, height=500)
fig.update_layout(mapbox_style="open-street-map")
fig.update_layout(margin={"r":0,"t":0,"l":0,"b":0})
fig.show()

<b>One can observe from the map, the particular rental places and its venues in detail under $7000/month. 
We could find an appartment with at the right price and in a location with desirable venues. The next step is to see if it is located near a subway metro station, in next cells work.

#### Mapping Manhattan Subway locations

<b>Manhattan subway metro locations ( address) was obtained from webscrapping sites such as Wikipedia, Google and NY Metro Transit. For simplification, a csv file was produced from the 'numbers' (Apple excel ) so that the reading of this file is the starting point here

Algorithm to find latitude and longitud for each subway metro station and add them to dataframe
This coding has been 'Markdown' just to simplify the file report, and the csv file will be read in cell below.
for n in range(len(mh)): address= mh['sub_address'][n] geolocator = Nominatim() location = geolocator.geocode(address) latitude = location.latitude longitude = location.longitude mh['lat'][n]=latitude mh['long'][n]=longitude

#print(n,latitude,longitude)
time.sleep(2)

print('Geodata completed')

save dataframe to csv file

In [119]:
# A csv file summarized the subway station and the addresses for next step to determine geodata
mh=pd.read_csv('datasets/NYC_subway_list.csv')
print(mh.shape)
mh.head()

(76, 2)


sub_station                              sub_address
0  Dyckman Street Subway Station   170 Nagle Ave, New York, NY 10034, USA
1       57 Street Subway Station                  New York, NY 10106, USA
2                       Broad St                  New York, NY 10005, USA
3             175 Street Station  807 W 177th St, New York, NY 10033, USA
4                 5 Av and 53 St                  New York, NY 10022, USA

In [120]:
mh=pd.read_csv('datasets/MH_subway.csv')
mh.head()

sub_station                              sub_address  \
0  Dyckman Street Subway Station   170 Nagle Ave, New York, NY 10034, USA   
1       57 Street Subway Station                  New York, NY 10106, USA   
2                       Broad St                  New York, NY 10005, USA   
3             175 Street Station  807 W 177th St, New York, NY 10033, USA   
4                 5 Av and 53 St                  New York, NY 10022, USA   

         lat       long  
0  40.861857 -73.924509  
1  40.764250 -73.954525  
2  40.730862 -73.987156  
3  40.847991 -73.939785  
4  40.764250 -73.954525

In [121]:
print(mh.shape)

(76, 4)


In [172]:
# removing duplicate rows and creating new set mhsub1
mhsub1=mh.drop_duplicates(subset=['lat','long'], keep="last").reset_index(drop=True)
print(mhsub1.shape)
mhsub1.head(10)

(22, 4)


sub_station  \
0                         175 Street Station   
1              Broadway-Lafayette St Station   
2                             Bowery Station   
3                        34th Street-Hudson    
4                   72 Street Subway Station   
5                                  Prince St   
6                                   Astor Pl   
7                          23 Street Station   
8  Brooklyn Bridge - City Hall Metro Station   
9            59 St - Columbus Circle Station   

                                    sub_address        lat       long  
0       807 W 177th St, New York, NY 10033, USA  40.847991 -73.939785  
1  E Houston St, New York, 10012, United States  40.725684 -73.997726  
2               Bowery, New York, NY 10002, USA  40.720217 -73.993720  
3     380-394 11th Ave, New York, NY 10001, USA  40.753038 -74.003860  
4              2nd Ave, New York, NY 10021, USA  40.766865 -73.959779  
5         88 Prince St, New York, NY 10012, USA  40.724412 -73.998317  
6        170 Nagle Ave, New York, NY 10034, USA  40.861857 -73.924509  
7            E 23rd St, New York, NY 10010, USA  40.740165 -73.986305  
8         18 Centre St, New York, NY 10007, USA  40.826342 -73.092191  
9         Columbus Cir, New York, NY 10023, USA  40.768387 -73.983178

<b> Lets map the location of the Manhattan Subway locations. On the other map lets map the subways and rental places.

In [130]:
fig = px.scatter_mapbox(mhsub1, lat= 'lat', lon = 'long', color = 'sub_station',labels = 'sub_station',
                        hover_data=['sub_station'],
                        color_discrete_sequence= px.colors.sequential.Viridis, 
                        zoom=10, height=500)
fig.update_layout(mapbox_style="open-street-map")
fig.update_layout(margin={"r":0,"t":0,"l":0,"b":0})
fig.show()

In [145]:
map_manhattan_rent = folium.Map(location=[latitude, longitude], zoom_start=13.3)

# add markers to map
for lat, lng, label in zip(mh_rent['Lat'], mh_rent['Long'],'$ ' + mh_rent['Rent_Price'].astype(str)+ mh_rent['Address']):      
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=6,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_manhattan_rent) 
    
    # add markers of subway locations to map
for lat, lng, label in zip(mhsub1['lat'], mhsub1['long'],  mhsub1['sub_station'].astype(str) ):
    label = folium.Popup(label, parse_html=True)
    folium.RegularPolygonMarker(
        [lat, lng],
        number_of_sides=6,
        radius=6,
        popup=label,
        color='red',
        fill_color='red',
        fill_opacity=2.5,
    ).add_to(map_manhattan_rent) 
    
    # Adds tool to the top right
from folium.plugins import MeasureControl
map_manhattan_rent.add_child(MeasureControl())

# Measurement ruler icon tool to measure distances in map
from folium.plugins import FloatImage
url = ('https://media.licdn.com/mpr/mpr/shrinknp_100_100/AAEAAQAAAAAAAAlgAAAAJGE3OTA4YTdlLTkzZjUtNDFjYy1iZThlLWQ5OTNkYzlhNzM4OQ.jpg')
FloatImage(url, bottom=5, left=85).add_to(map_manhattan_rent)

map_manhattan_rent

## RESULTS

<b> Lets Consolidate our findings of the cluster of venues, rental places and subway locations. 
    1. Bubbles are the clusters of venues
    2. Blue dots are apartments available for rent,
    3. Red - Subway stations

In [144]:
map_mh_one = folium.Map(location=[latitude, longitude], zoom_start=13.3)

# add markers to map
for lat, lng, label in zip(mh_rent['Lat'], mh_rent['Long'],'$ ' + mh_rent['Rent_Price'].astype(str)+ ', '+mh_rent['Address']):      
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=6,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_mh_one) 
    
    # add markers of subway locations to map
for lat, lng, label in zip(mhsub1['lat'], mhsub1['long'],  mhsub1['sub_station'].astype(str) ):
    label = folium.Popup(label, parse_html=True)
    folium.RegularPolygonMarker(
        [lat, lng],
        number_of_sides=6,
        radius=6,
        popup=label,
        color='red',
        fill_color='red',
        fill_opacity=2.5,
    ).add_to(map_mh_one) 


# set color scheme for the clusters
kclusters=5
x = np.arange(kclusters)
ys = [i+x+(i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(mh_merged['Latitude'], mh_merged['Longitude'], 
                                  mh_merged['Neighborhood'], mh_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=15,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1], legend = True,
        fill_opacity=0.7).add_to(map_mh_one)

    # Adds tool to the top right
from folium.plugins import MeasureControl
map_mh_one.add_child(MeasureControl())

# Measurement ruler icon tool to measure distances in map
from folium.plugins import FloatImage
url = ('https://media.licdn.com/mpr/mpr/shrinknp_100_100/AAEAAQAAAAAAAAlgAAAAJGE3OTA4YTdlLTkzZjUtNDFjYy1iZThlLWQ5OTNkYzlhNzM4OQ.jpg')
FloatImage(url, bottom=5, left=85).add_to(map_mh_one)

map_mh_one

### Selecting the Apartment.
After evalauting the rent, subway and venues. The two likely destinations would be.
1. West 105th Street	Upper West Side	1.89	4.0	2800	5300	40.799771	-73.966213
2. 60 Fulton Street	Rental in Financial District	3.09	2.0	2000	6185	40.708515	-74.005044

<b> Lets take a look at the Venues for the Aprtments 1 and 2.

<b>Apartment -1

In [154]:
## k_num is the cluster number to explore
k_num = 4
mh_merged.loc[mh_merged['Cluster Labels'] == k_num, mh_merged.columns[[1] + list(range(5, mh_merged.shape[1]))]]

Neighborhood 1st Most Common Venue 2nd Most Common Venue  \
2   Washington Heights                  Café                Bakery   
7          East Harlem    Mexican Restaurant                Bakery   
11    Roosevelt Island           Coffee Shop        Sandwich Place   
13      Lincoln Square               Theater  Gym / Fitness Center   
15             Midtown                 Hotel               Theater   
19        East Village        Ice Cream Shop                   Bar   
20     Lower East Side    Chinese Restaurant           Coffee Shop   
21             Tribeca   American Restaurant    Italian Restaurant   
22        Little Italy                Bakery                  Café   
25    Manhattan Valley           Coffee Shop                   Bar   
28   Battery Park City           Coffee Shop                  Park   
30       Carnegie Hill           Pizza Place           Coffee Shop   
33       Midtown South     Korean Restaurant           Coffee Shop   
37     Stuyvesant Town                   Bar                  Park   

        3rd Most Common Venue 4th Most Common Venue  \
2           Mobile Phone Shop           Pizza Place   
7   Latin American Restaurant         Deli / Bodega   
11                       Park   Japanese Restaurant   
13               Concert Hall                 Plaza   
15                Coffee Shop            Steakhouse   
19                   Wine Bar    Mexican Restaurant   
20                       Café                Bakery   
21                       Park                   Spa   
22                Yoga Studio          Cocktail Bar   
25                Pizza Place    Chinese Restaurant   
28                      Hotel             Wine Shop   
30             Cosmetics Shop                  Café   
33             Cosmetics Shop             Hotel Bar   
37              Boat or Ferry      Basketball Court   

        5th Most Common Venue 6th Most Common Venue 7th Most Common Venue  \
2              Sandwich Place                  Park                   Gym   
7             Thai Restaurant     French Restaurant                  Café   
11          Kosher Restaurant      Greek Restaurant        Baseball Field   
13         Italian Restaurant     French Restaurant                  Café   
15                 Food Truck          Cocktail Bar        Clothing Store   
19               Cocktail Bar           Pizza Place           Coffee Shop   
20  Latin American Restaurant                  Park          Cocktail Bar   
21                       Café              Boutique              Wine Bar   
22             Sandwich Place    Salon / Barbershop           Pizza Place   
25          Indian Restaurant    Italian Restaurant       Thai Restaurant   
28         Italian Restaurant         Women's Store                   Gym   
30                Yoga Studio                   Spa                   Bar   
33        Japanese Restaurant  Gym / Fitness Center                Bakery   
37                Coffee Shop        Baseball Field           Gas Station   

        8th Most Common Venue  9th Most Common Venue  \
2   Latin American Restaurant       Tapas Restaurant   
7                  Taco Place             Street Art   
11                        Gym  Outdoors & Recreation   
13                Opera House    Indie Movie Theater   
15                        Spa              Bookstore   
19         Chinese Restaurant              Speakeasy   
20        Japanese Restaurant            Pizza Place   
21                Coffee Shop       Greek Restaurant   
22             Ice Cream Shop     Seafood Restaurant   
25              Deli / Bodega     Mexican Restaurant   
28               Burger Joint                 Market   
30                  Bookstore      French Restaurant   
33               Cocktail Bar     Italian Restaurant   
37                Beer Garden        Harbor / Marina   

           10th Most Common Venue  
2              Mexican Restaurant  
7                      Steakhouse  
11                        Do

<b> Apartment -2

In [169]:
## k_num is the cluster number to explore
k_num = 3
mh_merged.loc[mh_merged['Cluster Labels'] == k_num, mh_merged.columns[[1] + list(range(5, mh_merged.shape[1]))]]

Neighborhood 1st Most Common Venue 2nd Most Common Venue  \
3               Inwood    Mexican Restaurant                Lounge   
5       Manhattanville         Deli / Bodega    Italian Restaurant   
10          Lenox Hill      Sushi Restaurant    Italian Restaurant   
12     Upper West Side    Italian Restaurant                   Bar   
16         Murray Hill        Sandwich Place                 Hotel   
17             Chelsea           Coffee Shop    Italian Restaurant   
18   Greenwich Village    Italian Restaurant      Sushi Restaurant   
27            Gramercy    Italian Restaurant            Restaurant   
29  Financial District           Coffee Shop                 Hotel   
31                Noho    Italian Restaurant     French Restaurant   
32        Civic Center  Gym / Fitness Center                Bakery   
35          Turtle Bay    Italian Restaurant           Coffee Shop   
36          Tudor City                  Café                  Park   
38            Flatiron    Italian Restaurant   American Restaurant   

     3rd Most Common Venue          4th Most Common Venue  \
3              Pizza Place                           Café   
5       Seafood Restaurant             Mexican Restaurant   
10             Coffee Shop           Gym / Fitness Center   
12                  Bakery  Vegetarian / Vegan Restaurant   
16     Japanese Restaurant           Gym / Fitness Center   
17          Ice Cream Shop                         Bakery   
18       French Restaurant                 Clothing Store   
27  Thrift / Vintage Store                   Cocktail Bar   
29                     Gym                      Wine Shop   
31            Cocktail Bar                      Gift Shop   
32      Italian Restaurant                   Cocktail Bar   
35              Steakhouse                       Wine Bar   
36             Pizza Place             Mexican Restaurant   
38                     Gym           Gym / Fitness Center   

   5th Most Common Venue          6th Most Common Venue 7th Most Common Venue  \
3               Wine Bar                         Bakery   American Restaurant   
5       Sushi Restaurant                    Beer Garden           Coffee Shop   
10           Pizza Place                   Burger Joint         Deli / Bodega   
12     Indian Restaurant                    Coffee Shop        Cosmetics Shop   
16           Coffee Shop             Salon / Barbershop          Burger Joint   
17             Nightclub                        Theater           Art Gallery   
18    Chinese Restaurant                           Café     Indian Restaurant   
27            Bagel Shop                    Coffee Shop           Pizza Place   
29            Steakhouse                            Bar    Italian Restaurant   
31             Bookstore                  Grocery Store    Mexican Restaurant   
32     French Restaurant                 Sandwich Place           Coffee Shop   
35      Sushi Restaurant                          Hotel          Noodle House   
36      Greek Restaurant               Sushi Restaurant                 Hotel   
38           Yoga Studio  Vegetarian / Vegan Restaurant                Bakery   

   8th Most Common Venue 9th Most Common Venue 10th Most Common Venue  
3                   Park    Frozen Yogurt Shop     Spanish Restaurant  
5     Falafel Restaurant            Bike Trail        Other Nightlife  
10                   Gym   Sporting Goods Shop        Thai Restaurant  
12              Wine Bar    Mexican Restaurant       Sushi Restaurant  
16     French Restaurant                   Bar     Italian Restaurant  
17    Seafood Restaurant   American Restaurant                  Hotel  
18                Bakery    Seafood Restaurant      Electronics Store  
27    Mexican Restaurant         Grocery Store              Wine Shop  
29           Pizza Place                  Park   Gym / Fitness Center  
31                 Hotel      Sushi Restaurant            Coffee Shop  
32                   Gym           

## DISCUSSION

It can be observed from the final map plot from the results section that we were able to compile the results accordingly under the trageted $7000/month. In a certain way comparing it to Dallas, its still a expensive place. Given the fact that the place would be shared by others too. It does not pose a problem. 

In the case of Apartment -1 it will be costing about $5300/ month. we are close to two subway stations 0.4 miles apart West 105th Street Upper West Side  and West 105th Street Upper West Side. Venues are located in cluster 4 and it is located in the Upper west side of Manhattan.

In the case Apartment -2 it will be costing about $6185/month. we are really close to the subway station 2 min away from Fulton Street Subway Station,Subway station,MTA. The place is located at Financial District. The venues belong to cluster 3.

Overall, we could go for Apartment-1 as it is much cheaper and we are saving more $1000 and the walking distance is decent. 

## CONCLUSIONS

1. I find it rewarding with the efforts, time and money spent. I believe this course with all the topics covered is well worthy of appreciation.
2. This project has shown me a practical application to resolve a real situation that has impacting personal and financial impact using Data Science tools.
3. The mapping with Folium is a very powerful technique to consolidate information and make the analysis and decision thoroughly and with confidence. I would recommend for use in similar situations.
4. One must keep open mind of the new tools for DS that continue to appear for application in several business fields.

## ACKNOWLEDGEMENT

1. Overall, I am impressed with the content and lab works presented during the Coursera IBM Certification Course.
2. This Capstone project presented me a great opportunity to practice and apply the Data Science tools and methodologies.
3. I have created a good project that I can present as an example to show my potential.
4. I believe I have acquired a good starting point to become a professional Data Scientist and I will continue exploring to creating examples of practical cases.